### ISSUE TITLE
Missing `characteristic(R::MPolyQuoRing)` method

### ISSUE THREAD

**Issue by fingolfin** [oscar-system/Oscar.jl#4239]
Presumably it would return `characteristic(base_ring(R))` *unless* the ideal being factored out happens to be the the entire ring (then return 1). So that would require a GB computation (but hopefully that already is being cached?)

**Comment by fieker** [comment]
On Thu, Oct 24, 2024 at 02:57:19AM -0700, Max Horn wrote:
> Presumably it would return `characteristic(base_ring(R))` *unless* the ideal being factored out happens to be the the entire ring (then return 1). So that would require a GB computation (but hopefully that already is being cached?)
> 
What is about Z[x,y]/<2>?
Or is this illegal by type?

**Comment by fingolfin** [comment]
Dunno. But even if we only support it for the case were the coeffs are from a field it would already be useful.

**Comment by joschmitt** [comment]
> What is about Z[x,y]/<2>? Or is this illegal by type?

One can construct the ring and do something with it:
```
julia> R, (x, y) = ZZ["x", "y"]
(Multivariate polynomial ring in 2 variables over ZZ, ZZMPolyRingElem[x, y])

julia> S, _ = quo(R, ideal(R, [2]))
(Quotient of multivariate polynomial ring by ideal (2), Map: R -> S)

julia> iszero(one(S))
false

julia> iszero(2*one(S))
true
```
Whether this works by design or by accident I don't know.

**Comment by fingolfin** [comment]
Thanks to @joschmitt for addressing this over fields. I guess for integers we could handle it by computing a GB and checking if it contains any degree 0 entries; if so, compute their gcd to get the characteristic.

Beyond that I guess it can get arbitrarily complicated and I'd be happy to throw the towel in the situations (as in, throw an error "if you want it, go ahead and implement it")

**Comment by thofma** [comment]
> Beyond that I guess it can get arbitrarily complicated and I'd be happy to throw the towel in the situations (as in, throw an error "if you want it, go ahead and implement it")

If the input is $I \subseteq R[x_1,...,x_n]$, one could do it generically by returning the characteristic of $R/I \cap R$. Generators for the intersection $R \cap I$ can be computed via a Gröbner basis as you described it.

**Comment by fingolfin** [comment]
Indeed. What I meant by "arbitrary complicated" is that then next we have to compute a quotient of `R`, and get its characteristic, and that can of course be a nasty quotient.

But in principle it shouldn't hard to write a generic method -- how well it works in practice is a separate question :-) (perhaps it should be an `@attr` method to ensure the computed value is remembered)

**Comment by JohnAAbbott** [comment]
In reply to @fingolfin my guess is that the computation of the G basis would likely dominate other costs (though one could probably construct counter-examples to my guess); so we likely do not need to worry about a quotient of `R` being nasty.

**Comment by fieker** [comment]
open for grabs... no further discussion now, unless there is a use case
